File: python/java_to_python/06-pgmAvion/src/co/edu/campusucc/poo/TipoMilitar.py
import time
from Avion import Avion  # assuming there is a class called Avion in avion.py

class TipoMilitar(Avion):
    def __init__(self):
        super().__init__()
        self.setTipo("Militar")

    def startingEngine(self):
        print("⛽Check Fuel ✅")
        for i in range(3):
            print("💥💥Spark💥💥")
            time.sleep(0.8 - (i * 0.4))
        print("Started the Motor...✈️✈️")

    def speedUp(self):
        speedUpString = "speed up in Progress: "
        for i in range(self.getVelocidad()):
            speedUpString += "✈️"

        for i in range(10):
            speedUpString += "✈️"
            print(speedUpString)
            time.sleep(0.5 - (i * 0.05))
        super().addSpeed(10)

    def stopingEngine(self):
        print("✅ Check the Turbine 🆗🆗")
        for i in range(2):
            print("⛔ turning off turbine❌❌")
            time.sleep(0.8 - (i * 0.4))
        print("Stoped...⚡⚡")

File: ruby/pgms_ruby/06-pgmAvion/TipoMilitar.rb
require_relative 'Avion'
require 'time'

class TipoMilitar < Avion
  def initialize
    super
    setTipo("Militar")
  end

  def startingEngine
    puts("⛽Check Fuel ✅")
    3.times do |i|
      puts("💥💥Spark💥💥")
      sleep(0.8 - (i * 0.4))
    end
    puts("Started the Motor...✈️✈️")
  end

  def speedUp
    speedUpString = "speed up in Progress: "
    getVelocidad.times do
      speedUpString += "✈️"
    end

    10.times do |i|
      speedUpString += "✈️"
      puts(speedUpString)
      sleep(0.5 - (i * 0.05))
    end
    super().addSpeed(10)
  end

  def stopingEngine
    puts("✅ Check the Turbine 🆗🆗")
    2.times do |i|
      puts("⛔ turning off turbine❌❌")
      sleep(0.8 - (i * 0.4))
    end
    puts("Stoped...⚡⚡")
  end
end